app/components/CalibratorActionForm.client.tsx

```diff
@@ -0,0 +1,150 @@
+import Form from "@rjsf/core";
+import {
+  BaseInputTemplateProps,
+  FieldTemplateProps,
+  getInputProps,
+  getSubmitButtonOptions,
+  ObjectFieldTemplateProps,
+  RJSFSchema,
+  SubmitButtonProps,
+} from "@rjsf/utils";
+import validator from "@rjsf/validator-ajv8";
+import { ChangeEvent, FocusEvent } from "react";
+
+function ObjectFieldTemplate(props: ObjectFieldTemplateProps) {
+  const { action_index } = props.schema;
+  return (
+    <div className="flex flex-col gap-4 flex-1">
+      <div className="flex flex-col gap-4">
+        <div className="card-title">{action_index + 1}.</div>
+        <p>{props.title}</p>
+      </div>
+      {props.properties.map((element) => {
+        return (
+          <div key={action_index} className="property-wrapper">
+            {element.content}
+          </div>
+        );
+      })}
+    </div>
+  );
+}
+
+function CustomFieldTemplate(props: FieldTemplateProps) {
+  const { classNames, style, help, description, errors, children } = props;
+  return (
+    <div className={`${classNames} flex flex-col gap-2`} style={style}>
+      {children}
+      {description}
+      {errors}
+      {help}
+    </div>
+  );
+}
+
+function BaseInputTemplate(props: BaseInputTemplateProps) {
+  const {
+    schema,
+    id,
+    options,
+    value,
+    type,
+    placeholder,
+    disabled,
+    readonly,
+    onChange,
+    onChangeOverride,
+    onBlur,
+    onFocus,
+    ...rest
+  } = props;
+  const onTextChange = ({
+    target: { value: val },
+  }: ChangeEvent<HTMLInputElement>) => {
+    // Use the options.emptyValue if it is specified and newVal is also an empty string
+    onChange(val === "" ? options.emptyValue || "" : val);
+  };
+  const onTextBlur = ({
+    target: { value: val },
+  }: FocusEvent<HTMLInputElement>) => onBlur(id, val);
+  const onTextFocus = ({
+    target: { value: val },
+  }: FocusEvent<HTMLInputElement>) => onFocus(id, val);
+
+  const inputProps = { ...rest, ...getInputProps(schema, type, options) };
+
+  return (
+    <input
+      className="input input-bordered max-w-xs"
+      id={id}
+      value={value}
+      placeholder={placeholder}
+      disabled={disabled}
+      readOnly={readonly}
+      onChange={onChangeOverride || onTextChange}
+      onBlur={onTextBlur}
+      onFocus={onTextFocus}
+      {...inputProps}
+    />
+  );
+}
+
+function CustomSubmitButton(props: SubmitButtonProps) {
+  const { uiSchema } = props;
+  const { norender } = getSubmitButtonOptions(uiSchema);
+  if (norender) {
+    return null;
+  }
+
+  return (
+    <div className="card-actions justify-end align-bottom">
+      <button className={`btn btn-primary`} type="submit">
```

Comment:
While clicking the button looks like it submits the form, I don't see any visible state change, so maybe the `className` needs to be toggled after submit? Do the props contain info to tell if this has happened?

Reply:
Yeah needs a backend change that i am working on, where the procedure records which actions have been completed. Once the backend has this state, the action will be "greyed out" on the front end, and the submit button disabled to indicate it's completed. In addition i intend on adding a "progress indicator", a display showing the current state of the procedure, a display showing current state of the transformer, and undo functionality.

WIP backend branch is here:
https://github.com/ssec-jhu/evolver-ng/tree/iainland/add-procedure-completed-actions-and-undo